code/modules/reagents/chemistry/recipes.dm

```diff
@@ -132,3 +277,102 @@
+	invert_reagents.my_atom = holder.my_atom //Give the gas a fingerprint
+	for(var/datum/reagent/reagent in holder.reagent_list) //make gas for reagents, has to be done this way, otherwise it never stops Exploding
+		if(!(reagent.type in required_reagents) || !(reagent.type in results))
+			continue
+		if(reagent.inverse_chem)
+			invert_reagents.add_reagent(reagent.inverse_chem, reagent.volume, no_react = TRUE)
+			holder.remove_reagent(reagent.type, reagent.volume)
+			continue
+		invert_reagents.add_reagent(reagent.type, reagent.volume, added_purity = reagent.purity, no_react = TRUE)
+		sum_volume += reagent.volume
+		holder.remove_reagent(reagent.type, reagent.volume)
+	if(invert_reagents.reagent_list)
+		smoke.set_up(invert_reagents, (sum_volume/10), holder.my_atom)
+		smoke.start()
+	holder.my_atom.audible_message("The [holder.my_atom] suddenly explodes, launching the aerosolized reagents into the air!")
+	if(clear_reactants)
+		clear_reactants(holder)
+	if(clear_products)	
+		clear_products(holder)
+
+//Spews out the contents of the beaker in a smokecloud
+/datum/chemical_reaction/proc/explode_smoke(datum/reagents/holder, datum/equilibrium/equilibrium, clear_products = TRUE, clear_reactants = TRUE)
+	var/datum/reagents/reagents = new/datum/reagents(2100, NO_REACT)//Lets be safe first
+	var/datum/effect_system/smoke_spread/chem/smoke = new()
+	reagents.my_atom = holder.my_atom //fingerprint
+	var/sum_volume = 0
+	for (var/datum/reagent/reagent in holder.reagent_list) 
+		if((reagent.type in required_reagents) || (reagent.type in results))
+			reagents.add_reagent(reagent.type, reagent.volume, added_purity = reagent.purity, no_react = TRUE)
+			holder.remove_reagent(reagent.type, reagent.volume)
+	if(reagents.reagent_list)
+		smoke.set_up(reagents, (sum_volume/10), holder.my_atom)
+		smoke.start()
+	holder.my_atom.audible_message("The [holder.my_atom] suddenly explodes, launching the aerosolized reagents into the air!")
+	if(clear_reactants)
+		clear_reactants(holder)
+	if(clear_products)	
+		clear_products(holder)
+
+//Pushes everything out, and damages mobs with 10 brute damage.
+/datum/chemical_reaction/proc/explode_shockwave(datum/reagents/holder, datum/equilibrium/equilibrium)
+	var/turf/T = get_turf(holder.my_atom)
+	holder.my_atom.audible_message("The [holder.my_atom] suddenly explodes, sending a shockwave rippling through the air!")
+	playsound(T, 'sound/chemistry/shockwave_explosion.ogg', 80, TRUE)
+	//Modified goonvortex
+	for(var/atom/movable/X in orange(3, T))
+		if(isliving(X))
+			var/mob/living/live = X
+			live.adjustBruteLoss(10)//Since this can be called multiple times
+		if(X.anchored)
+			continue
+		if(iseffect(X) || iscameramob(X) || isdead(X))
+			continue
+		var/distance = get_dist(X, T)
+		var/moving_power = max(4 - distance, 1)//Make sure we're thrown out of range of the next one
+		var/atom/throw_target = get_edge_target_turf(X, get_dir(X, get_step_away(X, T)))
+		X.throw_at(throw_target, moving_power, 1)
```

Comment:
Isn't this kind of thing already handled by the explosions subsystem?

Reply:
I didn't know that existed, I'll try to find it in the code and use that instead.

Reply:
Looking at it, the explosion subsystem is the kind that damages tiles and objects, right? This "explosion" is supposed to be a soft explosion that is more of a shockwave rather than damage - I created this "soft" explosion to have a similar effect, but to not kill, overly maim, or damage the aparatus. Should I add this code to the explosions subsystem instead?